src/Polytopes/Cone/properties.jl

```diff
@@ -161,49 +159,10 @@ isfulldimensional(C::Cone) = pm_cone(C).FULL_DIM
 ## Points properties
 ###############################################################################
 
-"""
-    rays_as_point_matrix(C::Cone)
-
-Return the rays of `C` as rows in a matrix.
-
-# Examples
-Here a cone is constructed from three rays. Calling `rays_as_point_matrix` reveals that one of these was redundant:
-```jldoctest
-julia> R = [1 0; 0 1; 0 2];
-
-julia> PO = positive_hull(R);
-
-julia> rays_as_point_matrix(PO)
-pm::Matrix<pm::Rational>
-1 0
-0 1
-```
-"""
-function rays_as_point_matrix(C::Cone)
-    pm_cone(C).RAYS
-end
-
-
-"""
-    facets_as_point_matrix(C::Cone)
-
-Return the facets of `C` as rows of a matrix.
-
-# Examples
-From this little example it is easy to see that the facets are displayed as their inside-pointing (w.r.t. the cone) normals.
-```jldoctest
-julia> R = [1 0; 1 1];
-
-julia> C = positive_hull(R);
-
-julia> facets_as_point_matrix(C)
-pm::Matrix<pm::Rational>
-0 1
-1 -1
-```
-"""
-facets_as_point_matrix(C::Cone) = pm_cone(C).FACETS
+# TODO: allow T = Cone when iterator can access `Cone(inequalities)``
+facets(as::Type{T}, C::Cone) where T<:Union{Halfspace} = HalfspaceIterator{as}(pm_cone(C).FACETS)
 
+facets(C::Cone) = facets(Halfspace, C)
```

Comment:
So for me this was weird:
```
julia> c = positive_hull([1 0; 0 1])
A polyhedral cone in ambient dimension 2

julia> facets(c)
2-element HalfspaceIterator{Halfspace}:
 The Halfspace of R^2 described by
1: x₁ ≦ 0

 The Halfspace of R^2 described by
1: x₂ ≦ 0
```
This is clearly wrong. Maybe this has something to do with that for cones there is no right hand side? Would it make sense to introduce `AffineHalfspace` for polyhedra and `LinearHalfspace` for cones?

Reply:
I can even continue:
```
julia> p = Polyhedron(facets(c))
A polyhedron in ambient dimension 2

julia> rays(p)
2-element VectorIterator{RayVector{Polymake.Rational}}:
 [0, -1]
 [-1, 0]
```

Reply:
I fixed the issue with `facets(::Cone)`. The sign had to be swapped. Of course I will include some tests (and a docstring with doctests) for the corresponding methods.

I have already thought about something like `AffineHalspace<:Halfspace` and `LinearHalfspace<:Halfspace`. We could have some more efficient definitions, e.g. not having to store the zero vector as `b`. `Halfspace` would have to be an abstract type then.

Reply:
Nevertheless I suggest going ahead as is now and do the Halfspace refactoring later. This PR touches many things and I need it for several other things that I want to do. And it is a huge improvement.